elastic/kibana
# [Fleet] Use latest compatible version in K8's manifest instead of latest available version

Fixes #175149 

## Summary

Uses the same "latest compatible version" logic present in `useAgentVersion` when we render agent install instructions when we display the Docker image version for K8's manifests.

## To test

1. Create a file at `x-pack/plugins/fleet/target/agent_versions_list.json` with some newer versions, e.g.

```json
[
  "8.16.0",
  "8.15.0"
]
```
2. Create an agent policy and add an integration policy for the `kubernetes` integration
3. Click the `Add Agent` button in the action dropdown for your agent policy
4. Observe the `image` value in the displayed K8's manifest has a version of `8.13.0` and not one of your newer versions

![image](https://github.com/elastic/kibana/assets/6766512/563dd3b3-3f5f-4c4c-94e9-5fb668173d16)

### Checklist
